### PR TITLE
BOAC-3252, If curated group modified then update student-count and SIDs of referencing cohorts

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-3252

Some "entropy reduction" in tests, too.

### CODE REVIEW

File: boac/api/curated_group_controller.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def curated_group_ids_per_sid(sid):
 
 @app.route('/api/curated_group/<curated_group_id>/remove_student/<sid>', methods=['DELETE'])
 @advisor_required
-@cross_origin(allow_headers=['Content-Type'])
+# @cross_origin(allow_headers=['Content-Type'])
```
Comment (lyttam) [inline boac/api/curated_group_controller.py]: @johncrossman Is this supposed to stay commented out?

Comment (johncrossman) [reply boac/api/curated_group_controller.py]: Whoops! I'll amend

**Comment by johncrossman** [comment]
Amended.

**Comment by lyttam** [comment]
Looks good!

**Pull request merged**